### PR TITLE
Prepare an environment to develop on MacOS by PyCharm

### PR DESCRIPTION
**Pull request by ditansu**
On MacOS we need:
1. escape colon in yaml by quotes 
2. use `sed -i ''`  instead `sed -i`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,6 @@ dist/
 htmlcov/
 node_modules/
 package-lock.json
+/src/editors.egg-info/
```
Comment (proofit404) [inline .gitignore]: How this was created?

File: tox.ini [hunk 2]
```diff
@@ -21,7 +21,7 @@ deps =
 commands =
   pre-commit autoupdate
   pre-commit run {posargs:--all-files}
-commands_post = sed -i 's/rev: .*$/rev: ""/g' .pre-commit-config.yaml
+commands_post = sed -i '' 's/rev: .*$/rev: ""/g' .pre-commit-config.yaml
```
Comment (proofit404) [inline tox.ini]: How does this work?

File: .gitignore (hunk 1 above)
Comment (ditansu) [reply .gitignore]: I don't know ))

File: tox.ini (hunk 2 above)
Comment (ditansu) [reply tox.ini]: On BSD-like OS If you use the -i option you need to provide an extension for your file(s) or '' instead.

**Comment by proofit404** [comment]
In my opinion, it would be better to split this PR into 3 different ones.

1. README file changes (gnu-sed) should go into the documentation of the development process. And we already have similar PR for this https://github.com/dry-python/mappers/pull/79
2. Exclude node modules from the flake8.
3. Quotes in the pre-commit config.

I think it would be better to use global Git ignore rules for stuff like `.idea` directory.

These changes should be applied to `dependencies`, `stories`, `mappers`, and `edittors`. That's why I propose to split them into different PRs.

Hope that makes sense.

Have a good day :tada: 

Best regards,
Artem.

**Comment by proofit404** [comment]
Due to my failure to document things, I would as you to rebase the branch and reword the commit message to pass commit lint util.

We follow [semantic-release](https://github.com/semantic-release/semantic-release#commit-message-format) commit messages format.

**Pull request closed without merging**